modules/core/common/createReduxStore.ts

```diff
@@ -8,11 +8,59 @@ export const getStoreReducer = (reducers: any) =>
     ...reducers
   });
 
-const createReduxStore = (reducers: Reducer, initialState: DeepPartial<any>, routerMiddleware?: Middleware): Store => {
+const requestMiddleware: Middleware = _state => next => action => {
+  const { types, APICall, ...rest } = action;
+  if (!types) {
+    return next(action);
+  }
+
+  const { REQUEST, SUCCESS, FAIL } = types;
+
+  next({ type: REQUEST || null, ...rest });
+
+  const handleAPICall = async () => {
+    try {
+      const result = await APICall();
+      const data = result && result.data;
+      next({
+        ...rest,
+        type: SUCCESS || null,
+        payload: data
+      });
+
+      return data;
+    } catch (e) {
+      if (e.response && e.response.data && e.response.data.status === 401) {
```

Comment:
Did you test this code? Are you sure the `status` is attached to `data`?

Reply:
Changed